test/executor.test.js

```diff
@@ -906,6 +906,62 @@ describe('executor', function() {
     });
   });
 
+  describe('when booting with lazy connect', function() {
+    var sampleInstruction = someInstructions({
+      dataSources: {
+        lazyConnector: {
+          connector: 'testLazyConnect',
+          name: 'lazyConnector',
+        },
+      },
+    });
+    var connectTriggered = true;
```

Comment:
@bajtos actually the variable will not reset lol. it's defined outside the hook, and stay the same value as previous test without reset.

But i think a more reasonable logic is 
```js
if (dataSource.settings.lazyConnect) {
     connectTriggered = false;
} else {
     connectTriggered = true;
}
```
then there is no need to reset it at the beginning.